### ISSUE TITLE
Is rank 1 decomposition possible with SVD seed?

### ISSUE THREAD

**Issue by omelnikov** [renozao/NMF#66]
NMF decomposition with rank 1 and `nndsvd` seed throws an error. Is this intended or is there a way to overcome the error? Here is an example:

```
> evar(nmf(USArrests, rank=1, seed='nndsvd'), as.matrix(USArrests))
Error in U[, i] : subscript out of bounds
> evar(nmf(USArrests, rank=1, seed='ica'), as.matrix(USArrests))
[1] 0.9731227
```

**Comment by omelnikov** [comment]
Update: even if rank-1 NMF seems to run, yielding expected variance, it still errors on plotting:
```
> basismap(nmf(USArrests, rank=1, seed='ica'))
Error in cut.default(x, breaks = breaks, include.lowest = T) : 
  'breaks' are not unique
> coefmap(nmf(USArrests, rank=1, seed='ica'))
Error in cut.default(x, breaks = breaks, include.lowest = T) : 
  'breaks' are not unique
> consensusmap(nmf(USArrests, rank=1, seed='ica'))
Error in cut.default(x, breaks = breaks, include.lowest = T) : 
  'breaks' are not unique
```

**Comment by renozao** [comment]
Which version are you running?
This might be fixed in the development version:

```R
devtools::install_github('renozao/NMF')
```

**Comment by omelnikov** [comment]
Thanks for reply. 
NMF: 0.20.6
R: 3.3.0
RStudio: 0.99.902
OS: Windows 7

I'll try the new version.

**Comment by omelnikov** [comment]
Running into dependencies issues with new NMF install. 
NMF 0.22 requires pkgmaker (>= 0.25.7), which requires repotools, which is not available for R 3.3.0.
I have not checked all dependencies.
Any suggestions?

**Comment by renozao** [comment]
pkgmaker does not require repotools, maybe ReportingTools, which is not required but suggested. But `devtools`  installs all dependencies. Maybe try with argument `vignette = FALSE`. And you need to install `pkgmaker` from the `develop` branch:

```R
source('http://bioconductor.org')
biocLite('ReportingTools')
library(devtools)
install_github('renozao/pkgmaker', ref = develop')
install_github('renozao/NMF')
```

**Comment by omelnikov** [comment]
Now I have `ReportingTools` v2.12.2,  `pkgmaker` v0.25.10. When trying `install_github('renozao/NMF')`  (with and without vignette), a message box pops up asking if I want to install additional build tools. Regardless of the answer, install fails:
```
> install_github('renozao/NMF') 
Downloading GitHub repo renozao/NMF@master
from URL https://api.github.com/repos/renozao/NMF/zipball/master
Error: Could not find build tools necessary to build NMF
```
I downloaded package archive file and tried manually:
```
> install.packages("C:/Users/Oleggio/Downloads/renozao-NMF-e46bec1.zip", repos = NULL, type = "win.binary")
Installing package into ‘C:/Users/Oleggio/Documents/R/win-library/3.3’
(as ‘lib’ is unspecified)
```

But, `NMF` is still showing version 0.20.6

**Comment by renozao** [comment]
You need to have all the utils to compile packages with C++ code. These are
provided by Rtools, available from the CRAN Windows page.

**Comment by omelnikov** [comment]
Yes, Rtools 3.3 have been installed all this time. I checked it first. It seems to be the latest version available on CRAN. Something else must be missing. Is there a log I can investigate?

**Comment by renozao** [comment]
The error `Error: Could not find build tools necessary to build NMF` suggests that R does not find Rtools. Have you checked that its libraries are in your PATH?

The zip file you downloaded contains the package source. You can unzip it and try to build it with:

```R CMD build path/to/NMF/directory```

You'll need to add R binary directory to your PATH to do that in the console.

**Comment by omelnikov** [comment]
Reinstalled `Rtools33.exe` and was able to successfully install `NMF` v0.22. However, the original error still persists:
```
> evar(nmf(USArrests, rank=1, seed='nndsvd'), as.matrix(USArrests))
Error in U[, i] : subscript out of bounds
> basismap(nmf(USArrests, rank=1, seed='nndsvd'))
Error in U[, i] : subscript out of bounds
```
Anything else I can give a try?
